File: COMP9041/lab06/whale.pl
#!/usr/bin/perl -w

if (@ARGV == 1) {
	$pod = 0;
	$counter = 0;
	$keyline = $ARGV[0];
	chomp($keyline);

	while (my $line = <STDIN>) {
		chomp($line);
		@line_split = split / /, $line, 2;
		$number = $line_split[0];
		$key = $line_split[1];
		if ($key eq $keyline) {
			$pod ++;
			$counter += $number;
		}
	}
	print "$keyline observations: $pod pods, $counter individuals\n"
}

File: COMP9041/lab06/whale.py
#!/usr/bin/python2.7

import sys
import re

key = sys.argv[1]
pod = 0
counter = 0
inputlines = []

for line in sys.stdin:
	line = re.sub(r" +", " ", line)
	line = re.sub(r"\n", "", line)
	sp = line.split(" ", 1)
	number = int(sp[0])
	name = sp[1]

	if (name == key):
		pod += 1
		counter += number

print (key + " observations: " + str(pod) + " pods, " + str(counter) + " individuals")
